src/backend/expungeservice/models/charge.py

```diff
@@ -34,6 +34,13 @@ def __post_init__(self):
         self.expungement_result = ExpungementResult(type_eligibility=type_eligibility, time_eligibility=None)
 
     def _build_type_eligibility(self):
+        type_eligibility = self._type_eligibility()
```

Comment:
I am wondering if this may cause some issues here. I am seeing below that this will raise a NotImplementedError. That is A-okay if the goal here is to use this as a base model that should not be implemented. Although, if the goal is to have your end users overwrite this method, then you might want to look at at using an Abstract Base Class @abstractmethod.